### ISSUE TITLE
Conflicting rule for *.test.ts files

### ISSUE THREAD

**Issue by lew** [gajus/eslint-config-canonical#42]
Your example config uses:
```
"extends": [
  "canonical/jest"
],
"files": "*.test.{ts,tsx}",
```
but in such a file the warning 
> Filename 'xyz.test.ts' does not match the naming convention.eslint(canonical/filename-match-regex)   

is triggered.

Maybe set the default regex to include ".test."

**Comment by gajus** [comment]
I think we should ignore all files that start with `.` or contain multiple `.`.

This will cover most files that trigger false-positives for this rule, e.g.

```
typegen.config.js
.gitlab-ci.yml
next-env.d.ts
.roarr.js
```

**Comment by gajus** [comment]
https://regex101.com/r/JsB7cL/1 I think this covers most of these cases.

It still requires camel case, but allows names to start with `.` and contain multiple `.` in the name.
